src/crewai/crew.py

```diff
@@ -255,15 +256,20 @@ def kickoff(
         i18n = I18N(prompt_file=self.prompt_file)
 
         for agent in self.agents:
+            # type: ignore # Argument 1 to "_interpolate_inputs" of "Crew" has incompatible type "dict[str, Any] | None"; expected "dict[str, Any]"
             agent.i18n = i18n
-            agent.crew = self
+            # type: ignore[attr-defined] # Argument 1 to "_interpolate_inputs" of "Crew" has incompatible type "dict[str, Any] | None"; expected "dict[str, Any]"
+            agent.crew = self  # type: ignore[attr-defined]
 
             if not agent.function_calling_llm:
                 agent.function_calling_llm = self.function_calling_llm
             if not agent.step_callback:
                 agent.step_callback = self.step_callback
-
-            agent.create_agent_executor()
+            # if using default crewai agent then run this
+            if isinstance(agent, Agent):
```

Comment:
This kind of logic is what I was trying to prevent with the introduction of a single agent interface to rule them all, you know.

Reply:
What about those customizations you've done based on the type of the agent?

Reply:
sorry scratch that. it was running the original agent. my changes are necessary for running the external agent properly.